docs/plugin/messaging.md

```diff
@@ -0,0 +1,886 @@
+                	"category": "Notification", // (2)
+                	"enabled": false
+                }
+            ]
+		},
+		"tracing": {
+			"settings": [
+				{
+					"category": "Fatal", // (3)
+					"enabled": true
+				},
+                {
+                    "module": "Plugin_SamplePlugin", // (4)
+                    "enabled": true
+                }
+			]
+		},
+		"reporting": {
+			"abbreviated": true,
+			"settings": [
+				{
+					"category": "TooLongWaitingForLock", // (5)
+					"enabled": true,
+                    "excluded": {
+                        "callsigns": [
+                            "com.example.SamplePlugin" // (6)
+                        ],
+                        "modules": [
+                            "Plugin_SamplePlugin" // (7)
+                        ]
+                    },
+                    "config": {
+                        "reportbound": 1000, // (8)
+                        "warningbound": 2000 // (9)
+                    }
+				}
+			]
+		}
+	}
+}
+```
+
+1. Reducing the amount of information in the messages coming from the `DirectOutput` (e.g. timestamp reduced to the time of day instead of the full date; removing file name, line number and class name for tracing type messages). This setting can be included separately in each message type.
+
+2. Disabling logging messages from the `Notification` category
+
+3. Enabling trace messages from the `Fatal` category
+
+4. Enabling **all** tracing categories used in the `SamplePlugin` plugin. The module name reflects the `MODULE_NAME` definition in the plugin `Module.h`
+
+5. Name of the category to configure
+
+6. Callsigns of plugins to exclude from the warning reporting category
+
+7. Module names to exclude from this warning reporting category
+
+8. Report bound indicating a value that must be exceeded for a report to be generated
+
+9. Warning bound indicating a value that must be exceeded for a warning to be generated (considered higher severity than a report). Note: warning bound should be >= report bound
+
+!!! tip
+	It is also possible to supply a filepath instead of an object to allow storing messaging configuration in a separate file
+	```json
+	{
+		"messaging": "/path/to/messagingconfig.json"
+	}
+	```
+
+Warning Reporting enables various runtime checks for potentially erroneous conditions and can be enabled on a per-category basis. These are typically time-based - i.e. a warning will be reported if something exceeded an allowable time. Each category can also have its own configuration to tune the thresholds for triggering the warning.
+
+!!! note
+	Warning Reporting is only available if Thunder is compiled with the `WARNING_REPORTING` option.
+
+## Runtime
+
+It is also possible to use the `MessageControl` plugin to edit the configuration values at runtime. At the moment it is possible to enable/disable any category at runtime, either globally for logging and warning reporting, or individually per plugin for the tracing messages.
```

Comment:
Can we show an example JSON-RPC request to change a messaging level at runtime?

Reply:
Sounds like yet another great idea, I didn't want to include a screenshot with the ThunderUI, since it seemed to be a bit too basic, but a JSON-RPC call is a great way to showcase that. I'll add this to the documentation.

Reply:
I have an example of documenting a JSON-RPC request here - personally quite like the arrow emojis to make it obvious which is the request and response :) https://rdkcentral.github.io/Thunder/plugin/config/#get-config